### PR TITLE
Filter outdated piped stat metrics

### PR DESCRIPTION
**Pull request by khanhtc1202**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/app/ops/pipedstatsbuilder/builder.go [hunk 1]
```diff
@@ -19,13 +19,18 @@ import (
...
+	"time"
 
 	"go.uber.org/zap"
 
 	"github.com/pipe-cd/pipe/pkg/cache"
 	"github.com/pipe-cd/pipe/pkg/model"
 )
 
+const (
+	liveStateExceedDuration = 2 * time.Minute
```
Comment (nakabonne) [inline pkg/app/ops/pipedstatsbuilder/builder.go:31]: nit: How about something like:
```suggestion
	pipedStatsRetention = 2 * time.Minute
```

Comment (nakabonne) [reply pkg/app/ops/pipedstatsbuilder/builder.go:31]: Looks like It isn't only for liveState. And `retention` is often used for how long the data is stored in the monitoring context.

Comment (khanhtc1202) [reply pkg/app/ops/pipedstatsbuilder/builder.go:31]: Thanks, lets me apply this 🙏

Comment (khanhtc1202) [reply pkg/app/ops/pipedstatsbuilder/builder.go]: Addressed by https://github.com/pipe-cd/pipe/pull/2281/commits/bfccaa3ec393d757f1d0d5282c135469a1d0dcd9 🙏

**Comment by nakabonne** [comment]
Nice
/approve

**Pull request merged**